src/serve_rendered.js

```diff
@@ -158,10 +158,7 @@
     // Iterate through paths, parse and validate them
     for (const providedPath of providedPaths) {
       // Logic for pushing coords to path when path includes google polyline
-      if (
-        providedPath.includes('enc:') &&
-        PATH_PATTERN.test(decodeURIComponent(providedPath))
-      ) {
+      if (providedPath.includes('enc:') && PATH_PATTERN.test(providedPath)) {
```

Comment:
do you think decodeURIComponent was added to prevent this CodeQL error?  I seem to recall seeing this in the original PR...

Reply:
looking more I am guessing that no this doesn't help this error.

Reply:
It would be good if we found a way to fix this. I wouldn't want to approve this PR with a high vulnerability

maybe some hints at https://codeql.github.com/codeql-query-help/javascript/js-polynomial-redos/

also this article makes me thing maybe decodeURIComponent was actually meant to help this  https://codeql.github.com/codeql-query-help/javascript/js-double-escaping/#recommendation

Reply:
I think the regex can improved. I did not manage to trigger the extreme behavior, but it's possible.
I pushed a change to the regex to try to improve its performance. Eslint still warns about "unsafe regular expression", but I believe it is a false positive.
Until you approve the CI workflow, I won't know for sure if CodeQL test pass.

I believe that removing `decodeURIComponent()` had nothing to do with the problem. It just made it more obvious to CodeQL that `providedPath` was user controlled.

Reply:
It looks like you are right, I confirmed this warning already exists in the current codebase
https://github.com/maptiler/tileserver-gl/security/code-scanning?query=Polynomial